OpenSim/Simulation/Model/Smith2016Ligament.h

```diff
@@ -0,0 +1,277 @@
+and the force-velocity curve need not be invertible.
+The rigid tendon formulation removes the singularities by ignoring the
+elasticity of the tendon. This assumption is reasonable for many muscles, but it
+is up to the user to determine whether this assumption is valid.
+The formulation that uses an elastic tendon with fiber damping removes
+singularities by solving the equilibrium equation with Newton's method. This is
+possible because the partial derivative of the equilibrium equation with respect
+to fiber velocity is always positive if \f$ \beta > 0\f$ and, thus, Newton's
+method can find a solution to the equilibrium equation.
+When either of these singularity-free formulations is selected, the minimum
+active-force-length value and the minimum permissible activation are set to
+zero. This is done as a convenience for the user, as these changes make the
+results of the model more realistic yet incur no performance penalty. The
+maximum pennation angle is left as acos(0.1) or 84.3 degrees, as allowing higher
+pennation angles results in an increasingly stiff fiber velocity state as
+pennation angle increases.
+
+<B>Reference</B>
+	Smith, C.R., Lenhart, R.L., Kaiser, J., Vignos, M.F. and Thelen, D.G.,
+	(2016). Influence of ligament properties on tibiofemoral mechanics
+	in walking. J Knee Surg, 29(02), 99-106.
+	
+	Wismans, J.A.C., Veldpaus, F., Janssen, J., Huson, A. and Struben, P., 
+	(1980). A three-dimensional mathematical model of the knee-joint. 
+	J Biomech, 13(8), 677-685.\n\n"
+
+	Blankevoort, L. and Huiskes, R., (1991).
+	Ligament-bone interaction in a three-dimensional model of the knee.
+	J Biomech Eng, 113(3), 263-269
+@author Colin Smith
+
+*/
+
+	class OSIMSIMULATION_API Smith2016Ligament : public Force  {
+		OpenSim_DECLARE_CONCRETE_OBJECT(Smith2016Ligament, Force)
+
+	public:
+	//=============================================================================
+	// PROPERTIES
+	//=============================================================================
+
+		OpenSim_DECLARE_UNNAMED_PROPERTY(GeometryPath,
+			"The set of points defining the path of the ligament")
+		OpenSim_DECLARE_PROPERTY(linear_stiffness, double,
+			"Slope of the linear portion of the force-strain curve of ligament")
+		OpenSim_DECLARE_PROPERTY(transition_strain, double,
+			"Strain at which ligament force-strain curve transitions from"
+			"quadratic to linear. Commonly 0.06 in literature.")
+		OpenSim_DECLARE_PROPERTY(normalized_damping_coefficient, double,
+			"Coefficient for normalized damping of ligament")
+		OpenSim_DECLARE_PROPERTY(slack_length, double,
+			"Length at which ligament begins developing tension")
+
+	//=============================================================================
+	// OUTPUTS
+	//=============================================================================
+
+		OpenSim_DECLARE_LIST_OUTPUT(dynamic_quantities, double, getDynamicQuantities, SimTK::Stage::Dynamics)
```

Comment:
Create separate outputs for each of these quantities.

Reply:
List outputs are for when the number of outputs is not fixed (e.g., a table might have any number of columns).